PMMain/Midas/MidasGUI/MidasThread.cpp

```diff
@@ -41,3 +42,97 @@ void MidasThread::threadEmitStateString(std::string str)
 {
     emit emitStateString(str);
 }
+
+
+std::vector<ringData> MidasThread::readKeyboardSetupFile()
+{
+    
+    std::ifstream readRingData("C:/Users/Owen/Documents/keyboardData.txt");
+    std::string temp;
+    int vect_num = 0;
+    int ringInOutSwitch = 0; //0 = ringIn , 1 = ringOut 
+    bool holdkey_check = false;
+
+    ringData::keyboard_value key_temp; //lvl1
+    std::vector<ringData::keyboard_value> tempRingInData[100]; //lvl2
```

Comment:
Why do you need to have 100 vectors for the inner ring? What does this represent? I feel like this should be a different data structure; 100 vectors seems excessive.